ui/src/pages/Logs/Logs.tsx

```diff
@@ -116,11 +117,33 @@ export default function Logs() {
         {}
       );
       setContainerIconColor(updatedContainerIconColor);
+
+      setElapsedTimeInMinutes(0);
     } catch (err) {
       console.error(err);
     }
   };
 
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setElapsedTimeInMinutes((prevElapsedTime) => prevElapsedTime + 1);
+    }, 60000);
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, []);
+
+  let message;
+  if (elapsedTimeInMinutes < 60) {
+    message = `Last refreshed ${elapsedTimeInMinutes}m ago`;
+  } else {
+    const elapsedHours = Math.floor(elapsedTimeInMinutes / 60);
+    const remainingMinutes = elapsedTimeInMinutes % 60;
+    const formattedTime = `${elapsedHours}h ${remainingMinutes}m`;
+    message = `Last refreshed ${formattedTime} ago`;
+  }
+
```

Comment:
There is a lot of logic being added to this page and I think it would be good to make this its own component, maybe "LogsLastRefreshedTime" or you can come up with a better name. You can pass a timestamp down to this component as a prop. Within this component you can do the setInterval. This is better than calling setInterval here in Logs, which will cause the entire page to re-render once per minute. Also, this file is getting pretty huge